api/v1alpha1/dataset_types.go

```diff
@@ -346,13 +348,28 @@ func (dataset *Dataset) GetDataOperationInProgress(operationType string) string
 // SetDataOperationInProgress set the data operation running on this dataset,
 func (dataset *Dataset) SetDataOperationInProgress(operationType string, name string) {
 	if dataset.Status.OperationRef == nil {
-		dataset.Status.OperationRef = map[string]string{}
+		dataset.Status.OperationRef = map[string]string{
+			operationType: name,
+		}
+		return
 	}
 
-	dataset.Status.OperationRef[operationType] = name
+	dataset.Status.OperationRef[operationType] = dataset.Status.OperationRef[operationType] + "," + name
 }
 
 // RemoveDataOperationInProgress release Dataset for operation
-func (dataset *Dataset) RemoveDataOperationInProgress(operationType string) {
-	delete(dataset.Status.OperationRef, operationType)
+func (dataset *Dataset) RemoveDataOperationInProgress(operationType, name string) string {
+	dataOpKeys := strings.Split(dataset.Status.OperationRef[operationType], ",")
```

Comment:
the `dataset.Status.OperationRef` can be nil

Reply:
done